src/cmdline_parser.c

```diff
@@ -30,7 +30,7 @@ bool rcutils_cli_option_exist(char ** begin, char ** end, const char * option)
 char * rcutils_cli_get_option(char ** begin, char ** end, const char * option)
 {
   size_t idx = 0;
-  size_t end_idx = end - begin;
+  size_t end_idx = (size_t)(end - begin);
```

Comment:
Looking around a bit, I think the basic problem here is that taking the difference between two pointers should be represented by `ptrdiff_t`, not `size_t` (in theory, the difference can be negative: https://bytes.com/topic/c/answers/460624-subtracting-pointers-type-use-store-result).  While changing to this, you'll probably also have to change the type of `idx` to match.